app/src/main/java/org/oppia/app/topic/review/TopicReviewFragmentPresenter.kt

```diff
@@ -6,82 +6,69 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.Observer
-import androidx.lifecycle.Transformations
 import androidx.recyclerview.widget.GridLayoutManager
 import org.oppia.app.databinding.TopicReviewFragmentBinding
+import org.oppia.app.databinding.TopicReviewSummaryViewBinding
 import org.oppia.app.fragment.FragmentScope
-import org.oppia.app.model.ProfileId
+import org.oppia.app.recyclerview.BindableAdapter
 import org.oppia.app.model.Subtopic
-import org.oppia.app.model.Topic
 import org.oppia.app.topic.PROFILE_ID_ARGUMENT_KEY
 import org.oppia.app.topic.RouteToReviewCardListener
 import org.oppia.app.topic.TOPIC_ID_ARGUMENT_KEY
-import org.oppia.domain.topic.TopicController
-import org.oppia.util.data.AsyncResult
-import org.oppia.util.logging.Logger
+import org.oppia.app.topic.review.reviewitemviewmodel.TopicReviewItemViewModel
+import org.oppia.app.viewmodel.ViewModelProvider
 import javax.inject.Inject
 
 /** The presenter for [TopicReviewFragment]. */
 @FragmentScope
 class TopicReviewFragmentPresenter @Inject constructor(
   activity: AppCompatActivity,
   private val fragment: Fragment,
-  private val logger: Logger,
-  private val topicController: TopicController
+  private val viewModelProvider: ViewModelProvider<TopicReviewViewModel>
 ) : ReviewSubtopicSelector {
   private var internalProfileId: Int = -1
   private lateinit var topicId: String
   private val routeToReviewListener = activity as RouteToReviewCardListener
 
-  private lateinit var reviewAdapter: ReviewSubtopicAdapter
-
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
     internalProfileId = fragment.arguments?.getInt(PROFILE_ID_ARGUMENT_KEY, -1)!!
     topicId = checkNotNull(fragment.arguments?.getString(TOPIC_ID_ARGUMENT_KEY)) {
       "Expected topic ID to be included in arguments for TopicReviewFragment."
     }
+    val viewModel = getTopicReviewViewModel()
+    viewModel.setInternalProfileId(internalProfileId)
+    viewModel.setTopicId(topicId)
+    viewModel.setReviewSubtopicSelector(this)
```

Comment:
Setting the listener this way is incorrect.

Instead you should inject the `activity` in your view model and write `private val routeToReviewListener = activity as RouteToReviewCardListener`  in viewmodel itself to use it.

Reply:
@rt4914 I have made the changes, but why is this method incorrect?

Reply:
This method is not incorrect, the approach is incorrect. In your implementation this how it is working:
Inject Activity in presenter -> create a `routeToReviewListener` object in Presenter -> Pass it to ViewModel -> Use `routeToReviewListener` in `ViewModel`.

And what I am suggesting is something which works like this:
Inject Activity in ViewModel -> create a `routeToReviewListener` object in VieModel -> Use `routeToReviewListener` in `ViewModel`.

The second approach makes it fairly easy.

Reply:
https://github.com/oppia/oppia-android/pull/849/files#r396470099